xds/internal/xdsclient/xdsresource/unmarshal_rds.go

```diff
@@ -147,6 +147,20 @@ func processClusterSpecifierPlugins(csps []*v3routepb.ClusterSpecifierPlugin) (m
 	for _, csp := range csps {
 		cs := clusterspecifier.Get(csp.GetExtension().GetTypedConfig().GetTypeUrl())
 		if cs == nil {
+			if csp.GetIsOptional() {
+				// "If a plugin is not supported but has is_optional set, then
+				// we will ignore any routes that point to that plugin"
+				cspCfgs[csp.GetExtension().GetName()] = nil
+				// Note, this conflates a not supported plugin with is optional
+				// set with a ClusterSpecifierPlugin that returns a nil
+				// BalancerConfig. However, due to this returned config being
+				// used as configuration for a child LB policy of the Cluster
+				// Manager LB Policy, also ignoring that route seems to make
+				// sense. The Cluster Specifier Plugin should also not return
+				// nil as the config in the first place.
```

Comment:
I don't think this comment is necessary.  A `nil` configuration is always invalid, and we can document it as such in the CSP package if it's not already and if you think it would help.

Reply:
Deleted this comment, added "A nil BalancerConfig is invalid" to cluster_specifier.go.